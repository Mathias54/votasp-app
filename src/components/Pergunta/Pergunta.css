.pergunta__title {
  margin: 0 0 20px 0;
  font-size: 1em;
  color: #333;
}

.pergunta__number {
  font-weight: bold;
}

.pergunta__lista-respostas .resposta-radio-btn__container {
  margin-bottom: 1em;
}

.pergunta__lista-respostas .resposta-radio-btn__container:last-child {
  margin-bottom: 0;
}
