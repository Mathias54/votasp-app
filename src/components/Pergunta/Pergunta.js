import React from 'react';
import { RESPOSTAS } from '../../constants/respostas';
import RespostaRadioButton from '../RespostaRadioButton/RespostaRadioButton';

import './Pergunta.css';

const Pergunta = ({ pergunta, responderQuestao, isAnswering, userAnswer }) => {
  return (
    <div className="pergunta__container">
      <p className="pergunta__title">
        <span className="pergunta__number">{pergunta.id}. </span>
        {pergunta.question}
      </p>
      <div className="pergunta__lista-respostas">
        {RESPOSTAS.map((resposta, index) => (
          <RespostaRadioButton
            key={resposta.value}
            id={pergunta.id}
            value={resposta.value}
            label={resposta.title}
            htmlFor={`${pergunta.id}_${index}`}
            onClick={responderQuestao}
            disabled={isAnswering}
            checked={userAnswer == resposta.value}
          />
        ))}
      </div>
    </div>
  );
};

export default Pergunta;
