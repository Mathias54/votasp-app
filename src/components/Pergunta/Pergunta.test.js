import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Pergunta from './Pergunta';

const defaultProps = {
  pergunta: {
    id: 1,
    question: 'Legalização do aborto para todos os casos.'
  }
}

describe('<Pergunta />', () => {
  it('should match snapshot', () => {
    const pergunta = shallow(<Pergunta {...defaultProps} />);

    expect(toJson(pergunta)).toMatchSnapshot();
  });
});
