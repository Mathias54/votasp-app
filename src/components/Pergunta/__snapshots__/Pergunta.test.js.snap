// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pergunta /> should match snapshot 1`] = `
<div
  className="pergunta__container"
>
  <p
    className="pergunta__title"
  >
    <span
      className="pergunta__number"
    >
      1
      . 
    </span>
    Legalização do aborto para todos os casos.
  </p>
  <div
    className="pergunta__lista-respostas"
  >
    <RespostaRadioButton
      htmlFor="1_0"
      id={1}
      key="-2"
      label="Discordo plenamente"
      value={-2}
    />
    <RespostaRadioButton
      htmlFor="1_1"
      id={1}
      key="-1"
      label="Discordo"
      value={-1}
    />
    <RespostaRadioButton
      htmlFor="1_2"
      id={1}
      key="1"
      label="Concordo"
      value={1}
    />
    <RespostaRadioButton
      htmlFor="1_3"
      id={1}
      key="2"
      label="Concordo plenamente"
      value={2}
    />
  </div>
</div>
`;
